File: 예진희/2 vsStudio_P/210208_cSharp_basic_P/2_class/2_class/Student2.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace _2_class
{   // 같은공간에 있는거(패키지)
    class Student2
    {
        // prop
        // 인스턴스 변수(속성)
        private string name;
        private string tel;

        // 클래스 변수 ( - : 인스턴스와 다른 공간에 잡힘, 객체 여러개 생성해도 하나만 참조)
        //public static string CLASS = "Java, C# 응용프로그램 개발자"; // * 클래스변수, 변경가능
        
        // 클래스 상수 변수
        public const string CLASS = "Java, C# 응용프로그램 개발자"; // ** static있느거 처럼 동작

        // 매개변수 존재하는 생성자**, ** ~ *** = 오버로딩
        public Student2()
        {
            name = "박길동";
            tel = "010-1111-1111";
            //기본값 초기화
        }

        // 매개변수 존재하는 생성자*** (많이 씀)
        public Student2(string name, string tel)
        {
            this.name = name;
            this.tel = tel;
        }

      

        // propfull (getter-setter 동시에, Name = 메소드/ 파스칼형식)
        public string Name {
            get { return name; }
            set { name = value; }
        } // value 내부에서 자동으로 만들어짐

        public string Tel {
            get { return tel; }
            set { tel = value; }
        }

        public void printStudentInfo() {
            Console.WriteLine("과정 : "+CLASS);
            //Console.WriteLine("과정 : " + Student2.CLASS); //공용이므로 private X

            Console.WriteLine("이름 : " + name);
            Console.WriteLine("전화 : " + tel);
            Console.WriteLine("----------------------");
        }
    }
}

File: 예진희/3 JavaClass/210106_class/class0106_1/src/Student2.java
//main이 없는 클래스파일 생성
//public은 자기 클래스 파일(.java)이 무조건 있어야함 (Student2.java 생성 됨)
public class Student2 {
	// 클래스 변수
	static String category = "통합응용SW과정";
	// private : 외부 클래스에서 접근 불가 /x 클래스 안에서만 접근가능, 캡슐화(정보를 외부로노출 x, 감싼다)
	// public : 외부 클래스에서 접근 가능
	// 인스턴스 변수
	private String name;
	private String tel;
	private String email;

	// 인스턴스 메소드
	// setter 값을 설정하는거
	public void setName(String name) {
		// x 매개변수로 정보전달 name 전(지역) / 뒤(지역)
		// x this: 객체생성된 위치정보 갖고있음, 인스턴스 변수에만 가능
		this.name = name;
		// x name = name1(지역)
	}

	// getter 정보를 가져옴
	public String getName() {
		return name;
	}

	public String getTel() {
		return tel;
	}

	public void setTel(String tel) {
		this.tel = tel;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}
}
